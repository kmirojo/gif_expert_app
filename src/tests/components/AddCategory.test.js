import React from "react";
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";
import "@testing-library/jest-dom";

describe('<AddCategory />', () => {

    const setCategories = jest.fn();
    
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(
            <AddCategory 
                setCategories={setCategories}
            />
        );
    });

    test('should match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should change textbox', () => {
        const input = wrapper.find("input");
        const value = "Hello World";

        input.simulate("change", {target: {value}});
        expect(wrapper.find("p").text().trim()).toBe(value);
    });
    
    test('should not post information in submit', () => {
        wrapper.find("form").simulate("submit", { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();
    });

    test('should call setCategories and clean textbox', () => {
        const value = "Hello World";

        wrapper.find("input").simulate("change", { target: { value } });
        wrapper.find("form").simulate("submit", { preventDefault(){} });

        expect( setCategories ).toHaveBeenCalled();
        expect(wrapper.find("input").props().value).toBe("");
    })
    
})
