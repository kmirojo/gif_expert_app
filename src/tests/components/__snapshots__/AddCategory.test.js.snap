// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory /> should match Snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
